lib/central_mail/upload_error.rb

```diff
@@ -23,10 +23,29 @@ class UploadError < StandardError
 
     STATSD_UPLOAD_FAIL_KEY = 'api.central_mail.upload.fail'
 
-    def initialize(message = nil, code: nil, detail: nil)
+    def self.extra_message_text(code, pdf_validator_options = {})
+      case code
+      when 'DOC106'
+        limit = pdf_validator_options[:size_limit_in_bytes] || \
+                PDFUtilities::PDFValidator::DEFAULT_OPTIONS[:size_limit_in_bytes]
+        "Limit is #{PDFUtilities::PDFValidator.formatted_file_size(limit)} per document"
+      when 'DOC108'
+        max_width = pdf_validator_options[:width_limit_in_inches] || \
+                    PDFUtilities::PDFValidator::DEFAULT_OPTIONS[:width_limit_in_inches]
+        max_height = pdf_validator_options[:height_limit_in_inches] || \
+                     PDFUtilities::PDFValidator::DEFAULT_OPTIONS[:height_limit_in_inches]
+        "Limit is #{max_width} in x #{max_height} in."
+      else
+        ''
+      end
+    end
```

Comment:
Yeah, totally, I'll add some specs!

Reply:
Updated - I'm really glad you asked (and kind of embarrassed) because I found some bugs as I was writing these tests - I'm more confident about this now!

Reply:
Thanks for doing that! We should have had them to begin with, and I'm glad we do now!